### PR TITLE
Using netlink to get client IP address in toolbox

### LINKED ISSUE #2688: Vch toolbox returns the wrong client interface address
Seen here: https://ci.vmware.run/vmware/vic/5678 (test 8-1-GuestTools)

The problem seems to be that we have `client` as an alias to the `external` interface; `ip link` output on the VCH:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: bridge: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether 00:0c:29:a4:82:70 brd ff:ff:ff:ff:ff:ff
3: external: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether 00:0c:29:a4:82:7a brd ff:ff:ff:ff:ff:ff
    alias client
```
Usually it is the opposite: external is aliased to client.

A potential fix would be to not use the Go apis to get the interface address, and use the [netlink](https://github.com/vishvananda/netlink) library.

### PR DESCRIPTION
**Pull request by hmahmood**
Fixes #2688

### CODE REVIEW

**Comment by anchal-agrawal** [comment]
lgtm

**Comment by andrewtchin** [comment]
lgtm

**Comment by dougm** [comment]
lgtm

File: lib/tether/toolbox.go [hunk 1]
```diff
@@ -219,25 +219,30 @@ func (t *Toolbox) halt() error {
 
 // externalIP attempts to find an external IP to be reported as the guest IP
 func (t *Toolbox) externalIP() string {
-	netif, err := net.InterfaceByName("client")
+	l, err := netlink.LinkByName("client")
```
Comment (dougm) [inline lib/tether/toolbox.go]: We should probably consider moving "client" to a constant, as its now referenced 6 times within this routine and likely many places elsewhere in vic.

**Comment by caglar10ur** [comment]
Lgtm

File: lib/tether/toolbox.go (hunk 1 above)
Comment (hickeng) [inline lib/tether/toolbox.go]: This should *not* be put in lib/portlayer if it's going to be referenced in lib/tether. That will again cause transitive inclusion of a vast amount of dead code.

We should consider moving this `vic-init` specific piece of logic into cmd/vic-init

**Comment by hickeng** [comment]
Opened this: https://github.com/golang/go/issues/17445

File: lib/tether/toolbox.go (hunk 1 above)
Comment (hmahmood) [inline lib/tether/toolbox.go]: @dougm @hickeng moved code out of toolbox. Will have a follow up PR to make `"client"` a constant.

**Comment by hmahmood** [comment]
:recycle:

**Pull request merged**